go-controller/pkg/cluster/gateway_init.go

```diff
@@ -176,12 +176,19 @@ func GatewayReady(nodeName string, portName string) (bool, error) {
 	// OpenFlow table 41 performs SNATing of packets that are heading to physical network from
 	// logical network.
 	for _, clusterSubnet := range config.Default.ClusterSubnets {
+		var cidr, match string
+		cidr = clusterSubnet.CIDR.String()
+		if strings.Contains(cidr, ":") {
+			match = "ipv6_src=" + cidr
+		} else {
+			match = "nw_src=" + cidr
+		}
 		stdout, _, err := util.RunOVSOfctl("--no-stats", "--no-names", "dump-flows", "br-int",
-			"table=41,ip,nw_src="+clusterSubnet.CIDR.String())
+			"table=41,"+match)
```

Comment:
You dropped "ip" here - that intentional?

Reply:
yes; it would have to be `ipv6` for IPv6, but it's unnecessary anyways because `nw_src` implies `ip` and `ipv6_src` implies `ipv6`

Reply:
thanks, lgtm then

Reply:
@danwinship I think we need that `ip` or `ipv6` to filter the flows. On my test machine,

Without ip, I get
```
# ovs-ofctl dump-flows br-int table=41,nw_src=192.168.0.0/16 |wc -l
40
``` 

and with `ip`, I get just one flow

```
[root@sdn-test1 ~]# ovs-ofctl dump-flows br-int table=41,ip,nw_src=192.168.0.0/16
 cookie=0xcb471b8d, duration=13762.513s, table=41, n_packets=0, n_bytes=0, priority=17,ip,metadata=0xb,nw_src=192.168.0.0/16 actions=ct(commit,table=42,zone=NXM_NX_REG12[0..15],nat(src=10.8.51.51))
```

Reply:
ah... yes, I noticed that it returned a bunch of non-matching flows but I assumed that the old code had done the same thing, since the next thing it does it to walk over the returned flows and check that there really is one containing the expected match...